### PR TITLE
Add new check in APIView to verify and enforce package name approval

### PR DESCRIPTION
**Pull request by praveenkuttappan**
Add new step in API check step to enforce package name approval for preview version release.

### CODE REVIEW

File: eng/common/scripts/Helpers/ApiView-Helpers.ps1 [hunk 1]
```diff
@@ -42,6 +42,11 @@ function Check-ApiReviewStatus($packageName, $packageVersion, $language, $url, $
     {
       Write-Host "API Review is approved for package $($packageName)"
     }
+    else if ($response.StatusCode -eq '202')
```
Comment (weshaggard) [inline eng/common/scripts/Helpers/ApiView-Helpers.ps1:45]: Do we need to update any callers of this method to ensure the call for prerelease packages as well?

File: eng/common/scripts/Helpers/ApiView-Helpers.ps1 [hunk 2]
```diff
@@ -42,6 +42,11 @@ function Check-ApiReviewStatus($packageName, $packageVersion, $language, $url, $
...
+    {
+      Write-Host "Package name $($packageName) is not yet approved by an SDK API approver. Package name must be approved to release a preview version if $($packageName) was never released as GA version."
+      Write-Host "You can check http://aka.ms/azsdk/engsys/apireview/faq for more details on package name Approval."
+    }
     else
```
Comment (weshaggard) [inline eng/common/scripts/Helpers/ApiView-Helpers.ps1:50]: Just to be cautious I would add an if StatusCode -eq 201 here and have a fallback else that represents all other failure cases. For example what happens if we get another error code that doesn't cause an exception?

File: eng/common/scripts/Create-APIReview.ps1 [hunk 3]
```diff
@@ -130,6 +130,13 @@ if ($packages)
             }
             elseif ($version.IsPrerelease)
             {
+                # Check if package name is approved. Preview version cannot be released without pacakge name approval
+                if ($respCode -eq '202' -and $pkgInfo.ReleaseStatus -ne "Unreleased")
```
Comment (weshaggard) [inline eng/common/scripts/Create-APIReview.ps1:134]: Below I see you added `!$pkgInfo.ReleaseStatus` which leads me to believe that there are cases where ReleaseStatus is null so might be worth adding that here too.

Comment (praveenkuttappan) [reply eng/common/scripts/Create-APIReview.ps1:134]: You are right. It's possible if there is no log for current version in change log.

File: eng/common/scripts/Create-APIReview.ps1 [hunk 4]
```diff
@@ -130,6 +130,13 @@ if ($packages)
             }
             elseif ($version.IsPrerelease)
             {
+                # Check if package name is approved. Preview version cannot be released without pacakge name approval
```
Comment (maririos) [inline eng/common/scripts/Create-APIReview.ps1:133]:
```suggestion
                # Check if package name is approved. Preview version cannot be released without package name approval
```

File: eng/common/scripts/Helpers/ApiView-Helpers.ps1 (hunk 1 above)
Comment (praveenkuttappan) [reply eng/common/scripts/Helpers/ApiView-Helpers.ps1:45]: Done

File: eng/common/scripts/Helpers/ApiView-Helpers.ps1 [hunk 5]
```diff
@@ -42,9 +42,19 @@ function Check-ApiReviewStatus($packageName, $packageVersion, $language, $url, $
     {
       Write-Host "API Review is approved for package $($packageName)"
     }
+    else if ($response.StatusCode -eq '202')
+    {
+      Write-Host "Package name $($packageName) is not yet approved by an SDK API approver. Package name must be approved to release a preview version if $($packageName) was never released as GA version."
```
Comment (weshaggard) [inline eng/common/scripts/Helpers/ApiView-Helpers.ps1:47]:
```suggestion
      Write-Host "Package name $($packageName) is not yet approved by an SDK API approver. Package name must be approved to release a beta version if $($packageName) was never released a stable version."
```

File: eng/common/scripts/Helpers/ApiView-Helpers.ps1 [hunk 6]
```diff
@@ -42,9 +42,19 @@ function Check-ApiReviewStatus($packageName, $packageVersion, $language, $url, $
...
+      Write-Host "You can check http://aka.ms/azsdk/engsys/apireview/faq for more details on package name Approval."
+    }
+    else if ($response.StatusCode -eq '201')
+    {
+      Write-Warning "API Review is not approved for package $($packageName). Release pipeline will fail if API review is not approved for a GA version release."
```
Comment (weshaggard) [inline eng/common/scripts/Helpers/ApiView-Helpers.ps1:52]:
```suggestion
      Write-Warning "API Review is not approved for package $($packageName). Release pipeline will fail if API review is not approved for a stable version release."
```

File: eng/common/scripts/Helpers/ApiView-Helpers.ps1 (hunk 5 above)
Comment (weshaggard) [reply eng/common/scripts/Helpers/ApiView-Helpers.ps1:47]: minor but we have been slowing trying to refer to our package releases as beta or stable as opposed to preview and GA.

File: eng/common/scripts/Create-APIReview.ps1 [hunk 7]
```diff
@@ -130,6 +130,13 @@ if ($packages)
             }
             elseif ($version.IsPrerelease)
             {
+                # Check if package name is approved. Preview version cannot be released without package name approval
+                if ($respCode -eq '202' -and $pkgInfo.ReleaseStatus -and $pkgInfo.ReleaseStatus -ne "Unreleased")
+                {
+                    Write-Host "Package name is not yet approved on APIView for $($PackageName). Package name must be approved by an API approver for a preview release if it was never released as GA version."
```
Comment (weshaggard) [inline eng/common/scripts/Create-APIReview.ps1:136]:
```suggestion
                    Write-Host "Package name is not yet approved on APIView for $($PackageName). Package name must be approved by an API approver for a beta release if it was never released a stable version."
```

**Review by weshaggard — APPROVED** [review]
A few minor naming things otherwise looks good.

**Pull request merged**